packages/app/src/cli/services/deploy/upload.ts

```diff
@@ -240,6 +237,27 @@ ${output.token.json(userErrors)}
   return res.data.functionSet.function?.id as string
 }
 
+async function uploadWasmBlob(extension: FunctionExtension, apiKey: string, token: string): Promise<string> {
+  const {url, headers, maxSize} = await getFunctionExtensionUploadURL({apiKey, token})
+  headers['Content-Type'] = 'application/wasm'
+
+  const functionContent = fs.readFileSync(extension.buildWasmPath())
+  const res = await http.fetch(url, {body: functionContent, headers, method: 'PUT'})
+  const resBody = res.body?.read()?.toString() || ''
+
+  if (res.status === 200) {
+    return url
+  } else if (res.status === 400 && resBody.includes('EntityTooLarge')) {
```

Comment:
`resBody.includes('EntityTooLarge')` seems very brittle. Should we define a more structured contract?